### PR TITLE
add event for custom cache keys

### PR DESCRIPTION
**Pull request by DevertNet**
### 1. Why is this change necessary?
There is no good way to manupulate the cache key. E.g. for custom prices etc.

### 2. What does this change do, exactly?
Added a event to allow the manipulation of the cache key hash.

### 3. Describe each step to reproduce the issue or behaviour.
-

### 4. Please link to the relevant issues (if any).
-

### 5. Checklist

- [ ] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [ ] I have written or adjusted the documentation according to my changes
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: src/Storefront/Framework/Cache/CacheStore.php [hunk 1]
```diff
@@ -205,6 +206,13 @@ private function generateCacheKey(Request $request): string
 
         $hash = 'md' . hash('sha256', $uri);
 
+        $event = new HttpCacheGenerateKeyEvent($hash, $request);
+        $this->eventDispatcher->dispatch(
+            $event,
+            CacheEvents::CACHE_GENERATE_KEY_EVENT
```
Comment (mitelg) [inline src/Storefront/Framework/Cache/CacheStore.php]: the second parameter is deprecated by Symfony. we should avoid it

Comment (DevertNet) [reply src/Storefront/Framework/Cache/CacheStore.php]: i removed the second parameter :)

**Comment by OliverSkroblin** [comment]
Hey @DevertNet,

thanks for the contribution. Please add some unit tests to ensure the new behavior.

**Comment by DevertNet** [comment]
Hey @OliverSkroblin,

i added the unit test. It only lasted hours, because i tested first the wrong cache sequence
 (╯°□°）╯ ︵ ┻━┻

**Comment by pantrtxp** [comment]
Hi @DevertNet, you still nee to accept the "Contributor License Agreement" bevor we can go on with this PR

**Comment by DevertNet** [comment]
@pantrtxp I accepted the CLA.

**Comment by pweyck** [comment]
Hi @DevertNet,

fixed by https://github.com/shopware/platform/pull/1090 .

Sorry that it took so long.

**Pull request closed without merging**